src/components/Choropleth.vue

```diff
@@ -13,212 +13,87 @@
-const {
-    selectedFeatures,
-    selectedIndicators,
-    selectedIndicator,
-    selectedCountryId,
-    appConfig,
-    countryBoundingBoxes,
-    admin0GeojsonFeature
-} = storeToRefs(useAppStore());
-
-useLoadingSpinner(map, waitingForMapBounds);
-
-const { getColour } = useColourScale(selectedIndicators);
-
+const { selectedFeatures, selectedIndicators, selectedIndicator, selectedCountryId, appConfig } =
+    storeToRefs(useAppStore());
 const featureProperties = appConfig.value.geoJsonFeatureProperties;
 
-const featureInSelectedCountry = (feature: Feature, selectedCountry) =>
+const { tooltipForFeature } = useTooltips();
+const { getFillAndOutlineColour } = useColourScale(selectedIndicators);
+
+const featureInSelectedCountry = (feature: Feature, selectedCountry: string) =>
     feature.properties[featureProperties.country] === selectedCountry;
+
 const getFeatureId = (feature: Feature) =>
     featureInSelectedCountry(feature, selectedCountryId.value)
         ? feature.properties![featureProperties.idAdm2]
         : feature.properties![featureProperties.idAdm1];
+
 const getFeatureName = (feature: Feature) =>
     featureInSelectedCountry(feature, selectedCountryId.value)
         ? feature.properties![featureProperties.nameAdm2]
         : feature.properties![featureProperties.nameAdm1];
 
-const getColourForFeature = (feature, indicator) => {
-    const featureId = getFeatureId(feature);
-    const featureIndicators = selectedIndicators.value[featureId];
-    return getColour(indicator, featureIndicators);
+const style = (f: Feature) => {
+    const isFaded = !!selectedCountryId.value && !featureInSelectedCountry(f, selectedCountryId.value);
+    const styleColors = getFillAndOutlineColour(selectedIndicator.value, getFeatureId(f), isFaded);
+    return { className: "geojson", ...styleColors };
 };
 
-// Useful for the e2e tests
-const dataSummary = computed(() => ({
-    "selected-indicator": selectedIndicator.value,
-    "selected-country-id": selectedCountryId.value,
-    "colour-scale": appConfig.value?.indicators[selectedIndicator.value]?.colourScale.name,
-    "feature-count": selectedFeatures.value.length,
-    "selected-country-feature-count": selectedFeatures.value.filter(
-        (f) => f.properties![featureProperties.country] === selectedCountryId.value
-    ).length,
-    bounds:
-        `S: ${bounds.value?.getSouth()} W: ${bounds.value?.getWest()} N: ${bounds.value?.getNorth()}` +
-        `E: ${bounds.value?.getEast()}`
-}));
+const getTooltip = (feature: Feature) => tooltipForFeature(getFeatureId(feature), getFeatureName(feature));
 
-const configureGeojsonLayer = (feature: Feature, layer: Layer) => {
-    const tooltip = tooltipForFeature(getFeatureId(feature), getFeatureName(feature));
-    if (tooltip) layer.bindTooltip(tooltip.content, tooltip.options);
-    layer.on({
-        click: async () => {
-            waitingForMapBounds.value = true;
+const layerOnEvents = (feature: Feature) => {
```

Comment:
I can see why you've kept this out of the useLeaflet composable, but it does feel a little bit backwards to passing this in, so I think a comment here about the context in which it's used would be good,

Reply:
done, yh i think this is good because it is kind of like an `@click` event that the normal geojson may expose, it also feels separate because anything to do with the loading or the url of the app should be the choropleths responsibility!